aws/resource_aws_eks_node_group.go

```diff
@@ -544,13 +545,13 @@ func resourceAwsEksNodeGroupDelete(d *schema.ResourceData, meta interface{}) err
 	}
 
 	if err != nil {
-		return fmt.Errorf("error deleting EKS Node Group (%s): %w", d.Id(), err)
+		return diag.Errorf("error deleting EKS Node Group (%s)", d.Id(), err)
 	}
 
-	_, err = waiter.NodegroupDeleted(conn, clusterName, nodeGroupName, d.Timeout(schema.TimeoutDelete))
+	_, err = waiter.NodegroupDeleted(ctx, conn, clusterName, nodeGroupName, d.Timeout(schema.TimeoutDelete))
 
 	if err != nil {
-		return fmt.Errorf("error waiting for EKS Node Group (%s) to delete: %w", d.Id(), err)
+		return diag.Errorf("error waiting for EKS Node Group (%s) to delete", d.Id(), err)
```

Comment:
Still need to pass `error`, just can't use `%w`.

```suggestion
		return diag.Errorf("error waiting for EKS Node Group (%s) to delete: %s", d.Id(), err)
```